cdc/processor.go

```diff
@@ -937,6 +946,23 @@ func (p *processor) addTable(ctx context.Context, tableID int64, replicaInfo *mo
 						lastResolvedTs = pEvent.CRTs
 						p.localResolvedNotifier.Notify()
 						resolvedTsGauge.Set(float64(oracle.ExtractPhysical(pEvent.CRTs)))
+
+						for {
+							if gTs := atomic.LoadUint64(&p.globalResolvedTs); gTs > lastResolvedTs {
```

Comment:
To save indentation, you could try if it needs break first.
```suggestion
							if gTs := atomic.LoadUint64(&p.globalResolvedTs); gTs <= lastResolvedTs {
								break
							}
```